pygly/alignment.py

```diff
@@ -165,15 +194,72 @@ def subtree_eq(self,a,b,root=True,mapids=False):
         if mapids:
             b.set_id(a.id())
 
-        for ii,jj in itermatchings(a.links(),b.links(),
-                                   lambda i,j: self.linkeq(i,j) and self.subtree_eq(i.child(),j.child(),root=False,mapids=mapids)):
+
+        assert len(filter(lambda l:l.repeat_bridge_link(), a.links())) in [0, 1]
+        alinks_no_repeat_path = a.links_without_repeat()
+        alinks_repeat_path    = a.links_with_repeat()
+
+        assert len(filter(lambda l:l.repeat_bridge_link(), b.links())) in [0, 1]
+        blinks_no_repeat_path = b.links_without_repeat()
+        blinks_repeat_path    = b.links_with_repeat()
+
+
+        for ii,jj in itermatchings(alinks_no_repeat_path, blinks_no_repeat_path, lambda i,j: self.linkeq(i,j) and self.subtree_eq_at_least_one(i.child(),j.child(),root=False,mapids=mapids,repeat_depth_a=repeat_depth_a,repeat_depth_b=repeat_depth_b)):
+            return True
+
+        limit = self.repeat_max_depth()
+        if repeat_depth_a > limit or repeat_depth_b > limit:
+            return False
+
+        repeat_match_a, repeat_match_b = False, False
+
+
+        for ii, jj in itermatchings(alinks_no_repeat_path, blinks_repeat_path,lambda i,j: self.linkeq(i,j) and self.subtree_eq_at_least_one(i.child(),j.child(),root=False,mapids=mapids,repeat_depth_a=repeat_depth_a,repeat_depth_b=repeat_depth_b+1)):
```

Comment:
Here, shouldn't there be an if condition on b having a bridge link? The assert above no withstanding, b.has_repeat_bridge_link in Monosaccharide would abstract the key decision...

Reply:
The if-condition was there. 
However, I made 2 functions for the Monosaccharide object, links_without_repeat and links_with_repeat. (line 95 and line 98 in pygly.Monosaccharide)
If the monosaccharide is not the end of the repeating unit, (which means this mono has no repeat bridge link), the method links_with_repeat returns an empty list. Because of that, the condition checking is not necessary any more.

Reply:
Argh. That's really hiding the logic - small helper methods like these should have clear and easily understood semantics - and it makes the alignment code harder to understand. You made two methods that work differently than the code they replaced, if I interpret correctly. Not ideal. Hmm.